src/Body.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Http\Client;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Allows special input as body
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+interface Body
+{
+    /**
+     * Returns a set of headers which is needed to correctly send the body
+     *
+     * Note: these headers get overwritten by headers manually passed to the client
+     *
+     * Content-Length is calculated automatically if possible
+     *
+     * @return array
+     */
+    public function getContentHeaders();
+
+    /**
+     * Convert data to a format which can be used to create a proper PSR-7 Stream
+     *
+     * @return string|StreamInterface
```

Comment:
Yes but it may conflict with some implementations like diactoros where the string is an url to stream and not a content. But maybe the string here as the same purpose ? In this case we should add documentation about the string return value.